R/parse-body.R

```diff
@@ -386,6 +386,16 @@ parser_json <- function(...) {
   })
 }
 
+#' @describeIn parsers GeoJSON parser. See [geojsonsf::geojson_sf()] for more details.
+#' @export
+parser_geojson <- function(...) {
+  parser_text(function(val) {
+    if (!requireNamespace("geojsonsf", quietly = TRUE)) {
+      stop("`geojsonsf` must be installed for `parser_geojson` to work")
+    }
```

Comment:
Let's move the require check earlier in the code so the error happens immediately, not when the parser is finally executed.

```suggestion
  if (!requireNamespace("geojsonsf", quietly = TRUE)) {
    stop("`geojsonsf` must be installed for `parser_geojson` to work")
  }
  parser_text(function(val) {
```